### PR TITLE
Update contribution relevant guides

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

Updates parts of the style-guide that we are not really enforcing, because they are unrealistically strict or we no longer agree with.

## How can this be tested?
Just reading.

## Checklist

- ~[ ] Unit tests have been updated/added~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Please include the following:
 - A meaningful PR title [guidelines](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#use-imperative-mood-in-your-commit-message-subject)
 - A summary of the change
 - Which issue is fixed (if there is one)
+  - Link to the Github issue or Jira ticket
```
Comment (waodim) [inline .github/pull_request_template.md:8]: Do we really want to post Jira links to tickets? Shouldn't this stay more intern?

File: doc/coding-style-guide.md [hunk 2]
```diff
@@ -21,11 +21,16 @@
...
+  - Shortnames for known Kubernetes Objects are fine. (`ns` for namespace)
+  - Avoid "stuttering". (In the `beepboop` package don't call you `struct` `BeepBoopController`, but just `Controller`)
+    - Relevant for: folder/package, file, struct, func and const/variable names.
+  - Do NOT shadow builtin names and packages.
 - Avoid using `client.Client` for 'getting' resources, use `client.Reader` (also known as `apiReader`) instead.
   - `client.Client` uses a cache (or tries to) that requires more permissions than normally, and can also give you outdated results.
-- Do not create methods with more than two parameters (in extremely rare occasions maybe three) except constructors and factory functions. Structs and interfaces exist for a reason.
+- Avoid creating functions with more than 3 params, except constructors and factory functions. Structs and interfaces exist for a reason.
 - Avoid returning responses (e.g., reconcile.Result, admission.Patched) in anything but Reconcile or Handle functions.
+- Run the linters locally before opening a PR, it will save you time.
```
Comment (andriisoldatenko) [inline doc/coding-style-guide.md:33]: May be it is worth mentioning that we have a pre-commit hook, which you can install via `make prerequisites/setup-pre-commit`

**Review by luhi-DT — APPROVED** [review]

File: doc/coding-style-guide.md (hunk 2 above)
Comment (0sewa0) [reply doc/coding-style-guide.md:33]: done

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Please include the following:
...
 - Relevant motivation and context
 
 ## How can this be tested?
```
Comment (albertogdd) [inline .github/pull_request_template.md:11]: Should we also remove the Checklist part? Is someone actually caring about it?

Comment (albertogdd) [reply .github/pull_request_template.md:11]: > We could shorten the current message/checkboxes, and move most of the things into the CONTRIBUTION.md and only have 1 checkbox that say "[ ] I followed the contribution guidelines". (I would actually prefer this)

I like this! It's just a pain whenever I create a PR without updating tests having to manually strikethrough the checkbox and so

**Review by albertogdd — APPROVED** [review]
Fair enough :+1:

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**